mrp_production_request/models/mrp_production_request.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+import openerp.addons.decimal_precision as dp
+from openerp.exceptions import UserError
+
+
+class MrpProductionRequest(models.Model):
+    _name = "mrp.production.request"
+    _description = "Manufacturing Request"
+    _inherit = "mail.thread"
+
+    @api.model
+    def _company_get(self):
+        company_id = self.env['res.company']._company_default_get(self._name)
+        return self.env['res.company'].browse(company_id.id)
+
+    @api.model
+    def _get_default_requested_by(self):
+        return self.env.user
+
+    @api.model
+    def create(self, vals):
+        """Add sequence if name is not defined and subscribe to the thread
+        the user assigned to the request."""
+        if not vals.get('name') or vals.get('name') == '/':
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'mrp.production.request') or '/'
+        res = super(MrpProductionRequest, self).create(vals)
+        if vals.get('assigned_to'):
+            res.message_subscribe_users(user_ids=[res.assigned_to.id])
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(MrpProductionRequest, self).write(vals)
+        for request in self:
+            if vals.get('assigned_to'):
+                self.message_subscribe_users(user_ids=[request.assigned_to.id])
+        return res
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.product_uom = self.product_id.uom_id
+        self.bom_id = self.env['mrp.bom']._bom_find(
+            product_id=self.product_id.id, properties=[])
+
+    @api.model
+    def _get_mo_valid_states(self):
+        return ['draft', 'confirmed', 'ready', 'in_production', 'done']
+
+    @api.one
```

Comment:
api.depends is widely use in conjunction with api.one in odoo standard. Do you think is really needed to avoid this usage of api.one?